### PR TITLE
add prisma

### PR DESCRIPTION
**Pull request by shamit6**

### CODE REVIEW

File: pages/api/balance-statuses.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+// Next.js API route support: https://nextjs.org/docs/api-routes/introduction
+import type { NextApiRequest, NextApiResponse } from 'next'
+import { PrismaClient, BalanceStatus } from '@prisma/client'
+
+const prisma = new PrismaClient()
```
Comment (deanshub) [inline pages/api/balance-statuses.ts]: Every file should do this? Should we move it to a separate single file module?

File: pages/api/balance-statuses.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  res: NextApiResponse<BalanceStatus | null>
+  ) {
+    try {
+      await prisma.$connect()
+      let response: BalanceStatus | null;
+      if (req.method === 'GET') {
+        response = await prisma.balanceStatus.findFirst();
+      } else if (req.method === 'PUT') {
+        const { amount } = req.body
+        response = await prisma.balanceStatus.create({
```
Comment (deanshub) [inline pages/api/balance-statuses.ts]: Is this upsert?

File: pages/transactions/index.tsx [hunk 3]
```diff
@@ -2,32 +2,61 @@ import { format } from 'date-fns'
...
+  const updateBalanceStatus = async (amount: number) => {
+    mutate(
+      '/api/balance-statuses?last=true',
+      { amount, createdAt: new Date() },
+      false
+    )
+    const data = await fetch(`/api/balance-statuses`, {
+      method: 'PUT',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ amount }),
```
Comment (deanshub) [inline pages/transactions/index.tsx:26]: The stringify is a must?

File: pages/api/balance-statuses.ts (hunk 2 above)
Comment (shamit6) [reply pages/api/balance-statuses.ts]: no, i want keeping every balance update

File: pages/transactions/index.tsx (hunk 3 above)
Comment (shamit6) [reply pages/transactions/index.tsx:26]: according to TS, yes

**Pull request merged**